x-pack/plugins/security_solution/public/management/cypress/e2e/mocks/artifacts_page.ts

```diff
@@ -0,0 +1,411 @@
+          value:
+            ' OSIS WindowsAND process.hash.*IS a4370c0cf81686c0b696fa6261c9d3e0d810ae704ab8301839dffd5d5112f476',
+        },
+      ],
+    },
+    update: {
+      formFields: [
+        {
+          type: 'clear',
+          selector: 'trustedApps-form-nameTextField',
+        },
+        {
+          type: 'input',
+          selector: 'trustedApps-form-nameTextField',
+          value: 'Trusted application name edited',
+        },
+        {
+          type: 'clear',
+          selector: 'trustedApps-form-descriptionField',
+        },
+        {
+          type: 'input',
+          selector: 'trustedApps-form-descriptionField',
+          value: 'This is the trusted application description edited',
+        },
+        {
+          type: 'click',
+          selector: 'trustedApps-form-conditionsBuilder-group1-entry0-field',
+        },
+        {
+          type: 'click',
+          selector: 'trustedApps-form-conditionsBuilder-group1-entry0-field-type-Path',
+        },
+        {
+          type: 'clear',
+          selector: 'trustedApps-form-conditionsBuilder-group1-entry0-value',
+        },
+        {
+          type: 'input',
+          selector: 'trustedApps-form-conditionsBuilder-group1-entry0-value',
+          value: 'c:\\randomFolder\\randomFile.exe, c:\\randomFolder\\randomFile2.exe',
+        },
+      ],
+      checkResults: [
+        {
+          selector: 'trustedAppsListPage-card-criteriaConditions',
+          value:
+            'OSIS WindowsAND file.pathis one of\nc:\\randomFolder\\randomFile.exe\nc:\\randomFolder\\randomFile2.exe',
+        },
+        {
+          selector: 'trustedAppsListPage-card-header-title',
+          value: 'Trusted application name edited',
+        },
+        {
+          selector: 'trustedAppsListPage-card-description',
+          value: 'This is the trusted application description edited',
+        },
+      ],
+    },
+    delete: {
+      confirmSelector: 'trustedAppsListPage-deleteModal-submitButton',
+      card: 'trustedAppsListPage-card',
+    },
+    pageObject: 'trustedApplications',
+    urlPath: 'trusted_apps',
+    emptyState: 'trustedAppsListPage-emptyState',
+  },
+  {
+    title: 'Event Filters',
+    pagePrefix: 'EventFiltersListPage',
+    create: {
+      formFields: [
+        {
+          type: 'input',
+          selector: 'eventFilters-form-name-input',
+          value: 'Event filter name',
+        },
+        {
+          type: 'input',
+          selector: 'eventFilters-form-description-input',
+          value: 'This is the event filter description',
+        },
+        {
+          type: 'click',
+          selector: 'fieldAutocompleteComboBox',
+        },
+        {
+          type: 'customClick',
+          selector: 'button[title="agent.type"]',
+        },
+        {
+          type: 'click',
+          selector: 'valuesAutocompleteMatch',
+        },
+        {
+          type: 'customClick',
```

Comment:
I would avoid having two different types for the same field. Also, having a function that uses cypress inside this file seems strange to me. We could add another field to determine if a selector must be used with the `cy.getBySel` (data-test-subj) or with the `cy.get`. But it will require us more conditions during the test.

Reply:
> Also, having a function that uses cypress inside this file seems strange to me.

Yeah, I think you are right.

> We could add another field...

This could be a good option! I don't think that having more conditions during the test is a problem: performance-wise it shouldn't be a problem, code readability-wise also okay, because we would separate the problem of `action.type` and the selector, so actually one additional `if` is needed, something like this:

```typescript
const runAction = (action: FormAction) => {
  let item;
  if (action.customSelector) {
    item = cy.get(action.customSelector);
  } else if (action.selector) {
    item = cy.getBySel(action.selector);
  }

  if (!item) return;

  if (action.type === 'click') {
    item.click();
  } else if (action.type === 'input') {
    item.type(action.value || '');
  } else if (action.type === 'clear') {
    item.clear();
  } else if (action.type === 'customClick') {
    item.click();
  } else if (action.type === 'removeItem') {
    item.click();
  }
};
```

What do you think?

Reply:
This looks great! I would remove the `if no item then return` as it's better to let the test fail if the desired item is not there. Will apply this change too. Thanks!